lib/modulesync/git.rb

```diff
@@ -18,7 +18,17 @@ def self.remote_branch_differ?(repo, local_branch, remote_branch)
         repo.diff("#{local_branch}..origin/#{remote_branch}").any?
     end
 
+    def self.default_branch(repo)
+      symbolic_ref = repo.branches.find { |b| b.full =~ %r{remotes/origin/HEAD} }
```

Comment:
This assumes you use origin as a remote. I always use ```upstream``` as a remote so I'd appreciate if that was overridable somehow but I do see that assumption is already present in a few places.

Reply:
You're right, it is hard coded in a number of places.  Shall we handle it in this PR or a future one?

Reply:
Since it's pre-existing code I'd handle it in a separate PR.